build-system/tasks/extension-helpers.js

```diff
@@ -394,7 +394,11 @@ function watchExtension(path, name, version, latestVersion, hasCss, options) {
       options.onWatchBuild(bundleComplete);
     }
   };
-  watch(`${path}/**/*`).on('change', debounce(watchFunc, watchDebounceDelay));
+  const fileExtensionToWatch = options.compileOnlyCss ? '.css' : '';
+  watch(`${path}/**/*${fileExtensionToWatch}`).on(
+    'change',
+    debounce(watchFunc, watchDebounceDelay)
+  );
```

Comment:
IIRC, we call `buildExtensions` twice during `gulp build`: Once for just CSS, and once for all JS. Will this change plus the change further down result in a double watch? (Can we test by first modifying a CSS file in an extension dir, and then modifying one of its JS files?)

Reply:
We do? I'm not seeing any double-rebuild behavior occurring even after testing out the exact scenario you described.

Reply:
Yep, we do during `gulp build`. You can test this by running `gulp build --watch`, waiting for a full build, and then editing a file.

First, we build the CSS for the runtime and all extensions, which appears to be necessary:

https://github.com/ampproject/amphtml/blob/b1a8300ac540823c097b58b30152a6dfb1bec494/build-system/tasks/css/index.js#L148

Then, we build the JS for all extensions:

https://github.com/ampproject/amphtml/blob/b1a8300ac540823c097b58b30152a6dfb1bec494/build-system/tasks/build.js#L78